javascript-challenges.test.js

```diff
@@ -14,30 +14,135 @@
 
 // a) Create a test with expect statements using the variables provided.
 
-const secretCodeWord1 = "Lackadaisical"
-// Expected output: "L4ck4d41s1c4l"
-const secretCodeWord2 = "Gobbledygook"
-// Expected output: "G0bbl3dyg00k"
-const secretCodeWord3 = "Eccentric"
-// Expected output: "3cc3ntr1c"
+function encodeString(input) {
+    const encodingMap = {
+      'a': '4',
+      'e': '3',
+      'i': '1',
+      'o': '0',
+    };
+  
+    // Use a regular expression to match and replace the characters
+    return input.replace(/[aeio]/g, (match) => encodingMap[match]);
+  }
+  
+  // Test cases using Jest
+  const secretCodeWord1 = "Lackadaisical";
+  const secretCodeWord2 = "Gobbledygook";
+  const secretCodeWord3 = "Eccentric";
+  
+  test('Encode secretCodeWord1', () => {
```

Comment:
While this setup for the test will work, we can consolidate this down into one describe block and it block where we test each individual test case.  Double check the setup and template in the Jest section of the syllabus.